### PR TITLE
feat(ratchet): proposeBaselineUpdate (H3)

### PR DESCRIPTION
**Pull request by camerontaylor**
## Goal
H3 proposeBaselineUpdate; breakdown: `ws-h-ratchets-policy.md` scope item 7; UC §6 row 64 (coverage auto-PR on main pattern); plan §5 op-table row "Ratchet ops".

## Stack
parent: `merge-queue` (H2 merged+promoted @ 7790720); children: `h4-selfhost-drills` (to be opened)

## Claims (acceptance checks this PR satisfies)
- [x] proposeBaselineUpdate: improvement → exactly one PR, idempotent on re-run — deterministic head branch (`ratchet/propose-<hash8>` over the canonical sorted (target, metric) set), `findOpenPrByHead` hit ⇒ the SAME PR is updated in place; pinned by the re-run test (still one PR, no second create).

## Deferred to
- The real gh-backed `BaselinePrEffects` implementation + `.github/workflows/ratchet-propose.yml` wiring (runs on push to main with CQ_AUTOMATION_TOKEN) → H4
- Barrel re-export (`src/index.ts` is lane I's) → phase-2 lane I; issue #66
- Hostile-adapter property-access residuals → issue #72; guard field-reconstruction residuals → issue #79

## Invariants exercised
I5: only a TIGHTEN against the committed baseline (direction always read from the committed file) is proposed — equal/looser produce no PR; missing/corrupt/non-regular baselines are skipped with reasons, never fabricated from nothing.
I1: kernel OpResult taxonomy — the op resolves to failed/indeterminate on effect faults (find rejection → failed; commit rejection → indeterminate, "may or may not have landed"); input is plain serializable data with arg-error guards incl. null/non-object input; the op never writes to the workspace (commits go through the effects seam).

## Token doctrine (recorded)
The PR is authored with `CQ_AUTOMATION_TOKEN`, never `GITHUB_TOKEN` — GitHub suppresses workflow runs on GITHUB_TOKEN-created PRs, so the required I4 check would be missing on the proposal PR. The op carries no tokens; `DEFAULT_PR_TOKEN` is the exported doctrine marker and the H4 effects implementation resolves the env var.

## Review rounds
round 1: FINDINGS 5 (all low) → all fixed (39624d2), plus 5 overlapping bot threads.
round 2: FINDINGS 5 (all low/info) → 4 fixed (be66192), 1 recorded (files[].path repo-relative contract note for H4's effects impl).
round 3 (cap): FINDINGS 7 (5 low / 2 info; invariants verified holding) → recorded, not fixed, per the below-medium protocol rule; ref-validation + verdict-derivation items deferred → issue #87 (batched with H4's effects pass). Review loop closed at the 3-round cap. **Best-effort merge with the round-3 findings recorded.**

## Review threads (bots)
8 threads (coderabbitai + chatgpt-codex-connector, incl. per-push re-review waves): 6 fixed (39624d2, be66192) / 0 dismissed / 2 deferred → issue #87; 0 unresolved (GraphQL-verified at merge time).

## Dismissed findings
(none)

## Recorded findings (round 3, below medium — not fixed, deferred/recorded per protocol)
- `proposal` verdict derives from the find result; `upsert.created` can disagree in the close-mid-flight race (prNumber/url already report the authoritative upsert identity) → issue #87
- `violatesRefRules` misses git's trailing-dot and `.lock`-component rules; the underscore exclusion is stricter than git (false rejects `release_1.0`) → issue #87
- test-depth: `containment.missing` all-skip branch and the leaf-symlink non-regular skip lack direct pins → issue #87
- seam contract for H4: `findOpenPrByHead` should disambiguate on (head, base) → issue #87
- digest-collision comment overclaims; wording should say "(target, metric) set" (value-differing sets collide by design — the head branch is per-set, which is the intended idempotency domain)

## Verification
- `npm run typecheck` → exit 0; `npm run lint` → exit 0
- `npx vitest run test/ops/ratchet` → 316/316 pass (7 files; 31 new in this PR)
- `npm run denylist` → findings only in untracked `.zcode/acp/sandbox.json` (agent-harness worktree state, never committed; CI scans a clean checkout)
- Known full-suite flake: `test/driver/*` 5s-timeout conformance tests under parallel load (pass in isolation; pre-existing, unrelated)

### CODE REVIEW

**Comment by camerontaylor** [comment]
## Review round 1 — H3 proposeBaselineUpdate (fresh reviewer)

Scope: `src/ops/ratchet/proposeBaselineUpdate.ts` (pass A) + `test/ops/ratchet/proposeBaselineUpdate.test.ts` (pass B). Verified `npx vitest run test/ops/ratchet` → 304/304 green. Invariant spot-checks that **passed**: `resolveBaselinesDir` is total (never rejects — captureBaseline.ts:96-121), so the unguarded await at proposeBaselineUpdate.ts:264 cannot leak a throw across the op seam; OpResult field names match kernel/types.ts:41-46; direction is always `baseline.direction` (proposeBaselineUpdate.ts:351,370); no path traversal (sanitizeSegment is `[a-z0-9-]`-only); **no token material crosses the op seam** — the op reads no env and carries no secrets, `DEFAULT_PR_TOKEN` is a literal doctrine marker. Acceptance-claim pinning is solid: exactly-one-PR (test:166-208, full outcome equality + one upsert + prs.size 1), re-run idempotency (test:231-252 — same number, `updated`, still one PR, upsertCalls 2), byte-exact head-hash convention incl. order-independence and set-difference (test:524-570). No high/medium findings; 5 low:

---

**1. [low] 'updated' outcome ignores the authoritative upsert result** — `src/ops/ratchet/proposeBaselineUpdate.ts:454-465`
When `findOpenPrByHead` hits, the outcome hardcodes `open.number`/`open.url` and never reads `upsert.created`/`upsert.number`. An effects impl that returns `created:true` with a *different* number (contract violation, or the found PR closed in the window between find and upsert) is misreported as `updated` against a stale number — the operator is told a PR was updated that was not. Not covered by the accepted #70/#72 residuals (those are read-race and property-access). Fix: report `upsert.number`/`upsert.url` in both branches (they describe what actually landed), keeping the `updated`/`created` verdict from the find.

**2. [low] Duplicate-pair dedup is unpinned and its comment overstates last-wins** — `src/ops/ratchet/proposeBaselineUpdate.ts:254-257`
"a later reading supersedes an earlier one" holds only among tightenings: for improvements `[{value:7},{value:12}]` against baseline 10, the later 12 is *skipped* ('not a tightening') and the earlier tighten stays applied — the outcome then lists the same (target, metric) in both `applied` and `skipped`. No test exercises duplicate (target, metric) improvements at all. Fix: correct the comment (winner = last qualifying tighten) and add a duplicate-pair test pinning the winning value, the single file entry, and the skip row for the looser reading.

**3. [low] `headPrefix` unvalidated → deterministic indeterminate loop on a bad prefix** — `src/ops/ratchet/proposeBaselineUpdate.ts:203-205,406`
Any git-ref-invalid prefix (space, `..`, leading `-`/`.`, `~`, control chars) deterministically produces an unusable branch name, so every run rejects at `commitAndUpsertPr` and is classified `indeterminate` ("may or may not have landed") forever even though nothing can ever land — a definitive misconfiguration reported as an unresolved fault. Fix: reject non-ref-safe prefixes in the arg guards as `failed` (e.g. `/^[A-Za-z0-9_][A-Za-z0-9._/-]*$/` plus no `..`/trailing `-`/`.lock`).

**4. [low] Raw target/metric interpolated into PR-body markdown** — `src/ops/ratchet/proposeBaselineUpdate.ts:415-417`
`z.string()` plus the arg guards allow newlines/backticks, so a committed `target` can inject arbitrary markdown lines into the proposal body — self-propagating, since bodies are authored from committed baselines this mechanism creates. Cosmetic only (I4 parses files, not the body; no secret can appear — nothing secret is in scope). Fix: strip control characters and escape backticks when composing the body bullet lines.

**5. [low] Test gap: most arg guards unpinned** — `test/ops/ratchet/proposeBaselineUpdate.test.ts:587-616`
The validation test covers null input, non-string ws, undefined base, non-finite value — but not `improvements` non-array (src:206), entry non-object (src:216), non-string target/metric (src:223), malformed `capturedAt` (src:240), non-string `headPrefix` (src:203), nor the empty-improvements → `none` fast path with zero effects calls. All are I1 boundary behavior (arg-error `failed` verdicts, nothing thrown). Fix: extend the validation test with these cases (table-driven is fine).

---

FINDINGS: 5

File: src/ops/ratchet/proposeBaselineUpdate.ts [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+        });
+        appliedByKey.set(relPath, {
+          path: relPath,
+          target: imp.target,
+          metric: imp.metric,
+          oldValue: baseline.value,
+          newValue: imp.value,
+          direction: baseline.direction,
+          content,
+        });
```
[the remark this answers is not in the document]
Comment (camerontaylor) [reply src/ops/ratchet/proposeBaselineUpdate.ts]: FIXED in 39624d2: duplicate (target, metric) improvements now collapse LAST-WINS BEFORE the tighten gate — a later looser reading supersedes and DROPS an earlier tightening (metric skipped, zero effects calls), a later tightening still proposes with the last value; the comment states exactly this; three tests pin the orderings (src/ops/ratchet/proposeBaselineUpdate.ts).

File: src/ops/ratchet/proposeBaselineUpdate.ts [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+        // readings — the improvement IS the reading).
+        if (tightens(baseline.value, imp.value, baseline.direction) === false) {
+          skipped.push({
+            target: imp.target,
+            metric: imp.metric,
+            reason:
+              `ratchet: metric '${imp.metric}' ${baseline.value} → ${imp.value} is not a ` +
+              `tightening (${baseline.direction}) — nothing to propose`,
+          });
+          continue;
```
[the remark this answers is not in the document]
Comment (camerontaylor) [reply src/ops/ratchet/proposeBaselineUpdate.ts]: FIXED in 39624d2: same last-wins-before-gate semantics — a superseded non-tightening duplicate leaves no record and the metric is not proposed; pinned by the [tighten, looser] → none test (src/ops/ratchet/proposeBaselineUpdate.ts).

File: src/ops/ratchet/proposeBaselineUpdate.ts [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+          value: {
+            proposal: 'updated',
+            prNumber: open.number,
+            prUrl: open.url,
+            head,
+            applied: appliedOut,
+            skipped,
+          },
+        };
+      }
```
[the remark this answers is not in the document]
Comment (camerontaylor) [reply src/ops/ratchet/proposeBaselineUpdate.ts]: FIXED in 39624d2: the outcome's prNumber/prUrl now come from the commitAndUpsertPr result whenever it carries one (the authoritative identity — a found PR can close between find and upsert), falling back to the find result only when the upsert reports none; pinned by the find=7/upsert=9 test (src/ops/ratchet/proposeBaselineUpdate.ts).

File: src/ops/ratchet/proposeBaselineUpdate.ts [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+        oldValue,
+        newValue,
+      }));
+      if (open !== null) {
+        return {
+          status: 'ok',
+          value: {
+            proposal: 'updated',
+            prNumber: open.number,
+            prUrl: open.url,
```
[the remark this answers is not in the document]
Comment (camerontaylor) [reply src/ops/ratchet/proposeBaselineUpdate.ts]: FIXED in 39624d2: same authoritative-identity change — the upsert's returned PR is what the outcome reports (src/ops/ratchet/proposeBaselineUpdate.ts).

File: src/ops/ratchet/proposeBaselineUpdate.ts [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+
+    // hash8 over the canonical JSON of the SORTED (target, metric) pairs —
+    // the same convention format.ts uses for baseline path digests (injective
+    // JSON encoding, deterministic). Same set ⇒ same head ⇒ idempotent
+    // upsert; different set ⇒ different head ⇒ its own proposal.
+    const pairs = applied.map((a): [string, string] => [a.target, a.metric]);
+    const hash8 = createHash('sha256')
+      .update(JSON.stringify(pairs), 'utf8')
+      .digest('hex')
+      .slice(0, 8);
```
[the remark this answers is not in the document]
Comment (camerontaylor) [reply src/ops/ratchet/proposeBaselineUpdate.ts]: FIXED in 39624d2: the improvement-set digest is widened from 8 to 12 hex (48 bits — the same width as the PR-61 path digest), so distinct improvement sets are far less likely to collide onto one head branch; pinned-hash tests updated (src/ops/ratchet/proposeBaselineUpdate.ts).

**Comment by camerontaylor** [comment]
## Round-2 review — proposeBaselineUpdate (H3)

**Scope:** Pass A `src/ops/ratchet/proposeBaselineUpdate.ts`, Pass B `test/ops/ratchet/proposeBaselineUpdate.test.ts` — current state @ 39624d2 (round-1 fixes verified in place). No skill/subagent used; reviewed directly.

**Verified holding:** I5 — only `tightens(baseline.value, imp.value, baseline.direction)` enters a proposal (src:402), direction always from the committed file, missing/corrupt/non-regular/escaped-dir/identity-disagreement all skip with reasons, and last-wins dedup runs **before** the gate so the newest reading governs (a later looser reading drops the metric entirely). I1 — find→`failed`, commit→`indeterminate`, arg-errors failed at the boundary; I traced every throw-capable call (`baselineRelPath`, `JSON.stringify` over validated string pairs, `join`, `renderBaseline`, `new Date()`) — nothing throws across the seam for structuredClone-safe inputs. Token doctrine intact (op carries no tokens; `DEFAULT_PR_TOKEN` is only a marker). `npx vitest run test/ops/ratchet/proposeBaselineUpdate.test.ts` → 28/28 pass locally.

### Findings

**1. [Low] `base` escapes both ref validation and body markdown-safety** — `src/ops/ratchet/proposeBaselineUpdate.ts:218` (validation) and `:472` (render). Round 1 gave `headPrefix` a git-ref guard and `target`/`metric` `mdSafe`, but `base` — which becomes the PR's base ref at the forge **and** renders raw inside backticks in the body (`Target branch: \`${input.base}\``) — is checked only `typeof === 'string'`. Any string passes: empty, backtick/newline-bearing (the exact body-injection surface mdSafe was added to close), or non-ref garbage that then surfaces as a forge effect-fault (`failed`/`indeterminate`) instead of the boundary arg-error the header promises.
*Fix:* run `input.base` through the same `HEAD_PREFIX_PATTERN` guard as `headPrefix` (ref-charset-valid ⇒ body-inert; also rejects the empty string).

**2. [Low] headPrefix guard admits refs git rejects** — `src/ops/ratchet/proposeBaselineUpdate.ts:233-237`. `ratchet//propose` and `ratchet/.hidden` pass the guard (pattern + `endsWith('/')` + `includes('..')` all satisfied) yet `git check-ref-format --branch` rejects the resulting heads (`//` = empty ref component; `/.x` = dot-leading component) — verified against git. The `-<digest>` suffix shields only the whole-ref rules (trailing `/`/`.`, `.lock`), so these component rules are the remaining holes; as written, the guard's own error message ("would form an invalid git ref") is incomplete and the invalid head becomes a doomed forge call.
*Fix:* also reject `headPrefix.includes('//')` and `/\/\./.test(headPrefix)` in the same guard.

**3. [Low] unit preservation is never exercised** — `test/ops/ratchet/proposeBaselineUpdate.test.ts:137` (`seedBaseline` never sets `unit`; it appears only in comments at :7 and :206). `unit: baseline.unit` at src:423 is uncovered: a regression stripping `unit` from every proposed baseline file passes the entire suite.
*Fix:* add one test seeding a unit-bearing baseline asserting the PR file bytes equal `renderBaseline({ …, unit: 'errors' })`.

**4. [Info] Weak clock-default capturedAt pin** — `test/ops/ratchet/proposeBaselineUpdate.test.ts:239` asserts only `Date.parse` non-NaN, which accepts strings the baseline's own strict parser (`isIso8601Instant`) would reject — the test wouldn't catch a regression loosening the clock default.
*Fix:* assert `expect(() => parseBaseline(content)).not.toThrow()` (strict capturedAt validation included).

**5. [Info] Seam contract leaves `files[].path` anchoring implicit** — `src/ops/ratchet/proposeBaselineUpdate.ts:112-124` (`commitAndUpsertPr` doc). Paths are the ws-virtual repo-relative form (`baselines/…`), but the interface never states that `ws` must be the repository checkout root for the H4 impl to commit them at the right place — evidence read from a nested `ws` would silently be proposed at repo-root paths.
*Fix:* one doc sentence on `commitAndUpsertPr`: "`files[].path` is repo-relative; `ws` is expected to be the repository checkout."

No High/Medium findings: no invariant violations, no security issues beyond the body-injection/consistency items above (containment, leaf-check-before-read, and digest collision width all check out; TOCTOU/#70, single-writer/#68, hostile-adapter residuals/#72 not re-raised).

FINDINGS: 5

File: src/ops/ratchet/proposeBaselineUpdate.ts [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+      if (typeof input.headPrefix !== 'string') {
+        return { status: 'failed', error: "ratchet: invalid input — 'headPrefix' must be a string" };
+      }
+      // A git ref, not free text: ref-name characters only, no leading or
+      // trailing slash, no '..' walk-up (the check-ref-format rules that
+      // matter for a branch prefix) — else the head could not exist at all.
+      if (
+        HEAD_PREFIX_PATTERN.test(input.headPrefix) === false ||
+        input.headPrefix.endsWith('/') ||
+        input.headPrefix.includes('..')
```
[the remark this answers is not in the document]
Comment (camerontaylor) [reply src/ops/ratchet/proposeBaselineUpdate.ts]: FIXED in be66192: violatesRefRules now rejects any '//' and any dot-leading path segment (plus the existing charset/edge rules) — 'ratchet//nightly' and 'ratchet/.hidden' both fail the arg-error boundary before any effects call; pinned in the headPrefix table (src/ops/ratchet/proposeBaselineUpdate.ts).

**Comment by camerontaylor** [comment]
## Round 3 review (final cap) — proposeBaselineUpdate (H3)

Reviewed the current state (`be66192`): Pass A `src/ops/ratchet/proposeBaselineUpdate.ts`, Pass B `test/ops/ratchet/proposeBaselineUpdate.test.ts`, cross-checked against the `format.ts` / `captureBaseline.ts` / `checkRatchet.ts` contracts. Verification: 31/31 tests pass locally (`npx vitest run test/ops/ratchet/proposeBaselineUpdate.test.ts`); every git ref-format claim below was verified with `git check-ref-format --branch` and by reproducing the predicate.

**Invariant check — holds.** Every `applied` entry requires a parsed, identity-matching baseline read from disk (containment → lstat-before-read → parse → identity → tighten gate, each with a named skip); direction always comes from the committed file (`tightens(baseline.value, imp.value, baseline.direction)`); last-wins dedup runs before the gate (superseded entries leave no record); find-reject → `failed` (nothing happened), upsert-reject → `indeterminate` (commit may have landed); no throw path crosses the op seam for structuredClone-safe inputs; the op never writes to the ws.

### Findings

**1. LOW — `proposal` verdict contradicts the upsert result in the find→upsert race** (carry-forward of the open thread; `src/ops/ratchet/proposeBaselineUpdate.ts:540`)
If the found PR closes before the upsert, `commitAndUpsertPr` returns `created: true` with a NEW PR's identity — the outcome then reports `proposal: 'updated'` next to the freshly created PR's number/url (pinned at test line 436); the inverse race (find miss, impl updated) reports `'created'` after an update. One-line fix: `proposal: upsert.created === true ? 'created' : open !== null ? 'updated' : 'created'` (and update the race test's pinned verdict).

**2. LOW — `violatesRefRules` misses two real git ref rules** (`src/ops/ratchet/proposeBaselineUpdate.ts:172`)
Verified against `git check-ref-format --branch`: `main.` (trailing dot — the open thread), `release.lock`, and `ratchet.lock/nightly` (component ending `.lock`) are all INVALID refs, yet all three pass the validator (predicate reproduced and confirmed); a garbage base/headPrefix then surfaces later as an effects-layer `indeterminate` instead of a boundary `failed` arg-error. One-line fix: also reject `value.endsWith('.')` and any segment ending `.lock`, with new rows in both ref tables.

**3. LOW — underscore exclusion refuses legitimate git branch names** (`src/ops/ratchet/proposeBaselineUpdate.ts:162`)
`release_1.0` and `main_v2` are VALID per `git check-ref-format` but are refused with the factually wrong wording "would form an invalid git ref" — a repo whose base branch contains an underscore can never be proposed against. One-line fix: allow `_` in `REF_PATTERN` (dropping the pinned `under_score` rows), or reword the error to name the deliberately stricter allowed subset.

**4. LOW — untested: the missing baselines-DIR path** (`test/ops/ratchet/proposeBaselineUpdate.test.ts:959`)
Every test that runs to completion seeds a baselines dir; the `containment.missing` branch (`src/ops/ratchet/proposeBaselineUpdate.ts:338-348` → all-skip with the I5 not-found wording) never executes under test — `checkRatchet.test.ts:363` pins its twin ("no dir, and dir present but empty"). One-line fix: add a bare-ws (no `baselines/` dir) test expecting skip-all with the not-found/I5 wording and zero effects calls.

**5. LOW — untested: symlink at the baseline leaf** (`test/ops/ratchet/proposeBaselineUpdate.test.ts:580`)
The non-regular-leaf skip is pinned only for the directory-squatter variant; the symlink-to-byte-valid-content-outside-the-ws variant that `checkRatchet.test.ts:508` pins is missing here, though the src comment claims the mirror. One-line fix: add a leaf-symlink case expecting the "not a regular file" skip.

**6. INFO — seam doc for H4: `findOpenPrByHead` keyed on head alone** (`src/ops/ratchet/proposeBaselineUpdate.ts:111`)
GitHub permits multiple open PRs from one head branch to different bases; the interface doc should require the impl to disambiguate on `(head, base)`, or the idempotent-update path can target the wrong PR. One-line fix: a doc line on the interface.

**7. INFO — comment overclaim on digest collisions** (`src/ops/ratchet/proposeBaselineUpdate.ts:472`)
"a different set cannot practically collide onto the same head branch" — improvement sets differing only in VALUES (same pairs) collide deterministically onto the same head; that's by design (the open PR is updated with the new values), but the claim (repeated in the header, line 33) should read "a different (target, metric) set". One-line fix: reword both comments.

**Verdict:** no blocking findings — I5/I1 hold end-to-end. Findings 1–3 are boundary/validation polish (both previously-open threads carried forward, now with empirical verification), 4–5 close the two mirror-test gaps, 6–7 are notes for H4.

FINDINGS: 7

File: src/ops/ratchet/proposeBaselineUpdate.ts [hunk 7]
```diff
@@ -0,0 +1,559 @@
...
+      // legitimately creates a DIFFERENT PR — its number/url is the truth.
+      // The find result is a FALLBACK only, for an impl that reports no
+      // identity from the upsert (the seam is wide; runtime is verified).
+      const prNumber = typeof upsert.number === 'number' ? upsert.number : (open?.number ?? null);
+      const prUrl =
+        typeof upsert.url === 'string' && upsert.url !== '' ? upsert.url : (open?.url ?? null);
+      return {
+        status: 'ok',
+        value: {
+          proposal: open !== null ? 'updated' : 'created',
```
[the remark this answers is not in the document]
Comment (camerontaylor) [reply src/ops/ratchet/proposeBaselineUpdate.ts]: DEFERRED to https://github.com/camerontaylor/cq-toolkit/issues/87 ([review-debt] #86 — ref-validation refinements + proposal-verdict derivation batched with H4's effects-implementation pass; the PR is at its 3-round review cap and the finding is recorded in the PR body's round-3 section).

File: src/ops/ratchet/proposeBaselineUpdate.ts [hunk 8]
```diff
@@ -0,0 +1,559 @@
...
+ * Anything violating this could not exist as a git ref at all — and could
+ * not ride clean markdown into the PR body either.
+ */
+function violatesRefRules(value: string): boolean {
+  return (
+    REF_PATTERN.test(value) === false ||
+    value.endsWith('/') ||
+    value.includes('//') ||
+    value.includes('..') ||
+    value.split('/').some((segment) => segment.startsWith('.'))
```
[the remark this answers is not in the document]
Comment (camerontaylor) [reply src/ops/ratchet/proposeBaselineUpdate.ts]: DEFERRED to https://github.com/camerontaylor/cq-toolkit/issues/87 ([review-debt] #86 — the ref-rules refinement class: trailing-dot and .lock components, underscore strictness; batched with H4's effects pass at the review cap; recorded in the PR body).

**Comment by camerontaylor** [comment]
all-clear: review r3 (cap) FINDINGS: 7 (5 low / 2 info) → recorded per the below-medium rule, ref-validation + verdict-derivation items deferred → issue #87 (batched with H4); threads: 8 total — 6 fixed / 0 dismissed / 2 deferred → #87; 0 unresolved (GraphQL-verified). Audit follows; this comment postdates the last commit be66192.
